pkgs/cli_config/lib/src/cli_provider.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'config.dart';
+import 'provider.dart';
+
+class CliProvider extends Provider {
+  /// Configuration options passed in via CLI arguments.
+  ///
+  /// Options can be passed multiple times, so the values here are a list.
+  ///
+  /// Stored as a flat non-hierarchical structure, keys contain `.`.
+  final Map<String, List<String>> _cli;
+
+  CliProvider(this._cli);
+
+  @override
+  String? getOptionalString(String key) {
```

Comment:
https://dart.dev/guides/language/effective-dart/design#avoid-starting-a-method-name-with-get

Reply:
I would use `read`, or consider dropping the entire verb and using `boolean` instead of `bool`? `read` is only one more character.

I don't ultimately feel super strongly for this particular case, but I do care that Dart owned/maintained packages  follow effective Dart guidelines, so we don't set an example that is contrary to effective Dart. Ultimately people will imitate the types of apis we put out there, and use them as justification for their own, unaware of any deeper reasoning. In fact `shared_preferences` is a perfect example of this, in this exact thread...

Reply:
I'd drop the verb. Changing `get` to `read`, which still has no *semantic meaning* different from "get", "fetch", "lookup" or "find", is not an improvement.

Don't use words that say nothing. Document and name getter-like methods like get getters (noun phrases).
That's what this is.

I'd use `string`, `stringValue`, `stringValueOf` or (more dubious) `stringOf`.

For `bool` and `int`, where we may don't want to get naming conflicts (we can, they are *not* reserved words, they just shadow the core types), we can use `boolean` and `integer`, or `boolValue`/`intValue`.

Or just use `int` and `bool` and deal with the conflicts:
```dart
import "dart:core";
import "dart:core" as core;

....

   core.bool bool(String key) { ... }

```
(You need to use `core.bool` everywhere in the class, highly annoying.)